src/components/axis.ts

```diff
@@ -1,33 +1,53 @@
 import { axisRight, axisBottom } from 'd3-axis';
 import { Selection } from 'd3-selection';
-import { ScaleLinear } from 'd3-scale';
+import { ScaleLinear, scaleLinear } from 'd3-scale';
 import { BaseType } from 'd3';
-import { OnResizeEvent } from '../interfaces';
+import { OnResizeEvent, OnRescaleEvent } from '../interfaces';
+
+type Options = {
+  offsetX: number;
+  offsetY: number;
+};
 
 export class Axis {
-  mainGroup: Selection<SVGElement, any, null, undefined>;
-  scaleX: ScaleLinear<number, number>;
-  scaleY: ScaleLinear<number, number>;
+  private mainGroup: Selection<SVGElement, any, null, undefined>;
+  private _scaleX: ScaleLinear<number, number>;
+  private _scaleY: ScaleLinear<number, number>;
   showLabels = true;
   labelXDesc: string;
   labelYDesc: string;
   measurement: string;
```

Comment:
What is measurement, should it me unit instead?

Reply:
sure, or unitOfMeasure